tasks/eva_3667/src/main/groovy/eva3667/remediate_lowercase_nucleotide.groovy

```diff
@@ -0,0 +1,362 @@
+        ]
+
+        Map<String, Set<String>> uppercaseHgvs = getUppercaseHgvs(lowercaseVariant)
+        String uppercaseHgvsName = uppercaseHgvs.values().iterator().next()
+        boolean hgvsNameAlreadyInDB = variantInDB.getHgvs().stream()
+                .map(hgvs -> hgvs.getName())
+                .anyMatch(name -> name.equals(uppercaseHgvsName))
+        if (!hgvsNameAlreadyInDB) {
+            HgvsMongo hgvsMongo = new HgvsMongo(uppercaseHgvs.keySet().iterator().next(), uppercaseHgvsName)
+            updateOperations.add(Updates.push("hgvs", new Document("\$each", Collections.singletonList(
+                    mongoTemplate.getConverter().convertToMongoType(hgvsMongo)))))
+        }
+
+        logger.info("case merge all sid fid are different - updates: {}", updateOperations)
+        mongoTemplate.getCollection(VARIANTS_COLLECTION).updateOne(Filters.eq("_id", newId),
+                Updates.combine(updateOperations))
+        logger.info("case merge all sid fid are different - delete lowercase variant: {}", lowercaseVariant)
+        mongoTemplate.remove(Query.query(Criteria.where("_id").is(lowercaseVariant.getId())), VARIANTS_COLLECTION)
+
+        remediateAnnotations(lowercaseVariant.getId(), newId)
+    }
+
+    void remediateCaseCantMerge(String workingDir, Set<Pair> sidFidPairsWithGtOneEntry, String dbName, VariantDocument lowercaseVariant) {
+        logger.info("case can't merge variant - processing variant: {}", lowercaseVariant)
+
+        String nmcDirPath = Paths.get(workingDir, "non_merged_candidates").toString()
+        File nmcDir = new File(nmcDirPath)
+        if (!nmcDir.exists()) {
+            nmcDir.mkdirs()
+        }
+        String nmcFilePath = Paths.get(nmcDirPath, dbName + ".txt").toString()
+        try (BufferedWriter nmcFile = new BufferedWriter(new FileWriter(nmcFilePath, true))) {
+            for (Pair<String, String> p : sidFidPairsWithGtOneEntry) {
+                nmcFile.write(p.getFirst() + "," + p.getSecond() + "\n")
+            }
+        } catch (IOException e) {
+            logger.error("error writing case variant can't be merged in the file:  {}", lowercaseVariant)
+        }
+    }
+
+    void remediateCaseMergeAllCommonSidFidHasOneFile(VariantDocument variantInDB,
+                                                     VariantDocument lowercaseVariant, Set<Pair> sidFidPairNotInDB, String newId) {
+        logger.info("case merge all common sid fid has one file - processing variant: {}", lowercaseVariant)
+
+        Set<VariantSourceEntryMongo> candidateFiles = lowercaseVariant.getVariantSources().stream()
+                .filter(vse -> sidFidPairNotInDB.contains(new Pair(vse.getStudyId(), vse.getFileId())))
+                .collect(Collectors.toSet())
+        Set<VariantStatsMongo> candidateStats = lowercaseVariant.getVariantStatsMongo().stream()
+                .filter(vse -> sidFidPairNotInDB.contains(new Pair(vse.getStudyId(), vse.getFileId())))
```

Comment:
Trying to remember the logic for this case, why we don't need to check the entire file/stats sub-document and don't need to combine these sub-documents ever...
Is it because this must arise from the same file being loaded before & after the pipeline fix? And there can't be duplicates in the original file with different casing (e.g. `1_123_A_T` and `1_123_A_t`)

Reply:
As discussed in the standup, merging is fine but we will recalculate the stats